Taking variant classification into account when filtering hotspots

onursumer opened the pull request:
Improved cancer hotspot service so that we better filter hotspots for a given transcript.

Needed to move over and refactor some code from `genome-nexus-annotation-pipeline`. Eventually we should move everything there to genome nexus.

onursumer commented on service/src/main/java/org/cbioportal/genome_nexus/service/annotation/VariantClassificationResolver.java:
```diff
@@ -0,0 +1,163 @@
+        if (transcriptConsequence != null)
+        {
+            variantClassification = this.resolveVariantClassification(
+                this.consequencePrioritizer.pickHighestPriorityConsequence(transcriptConsequence.getConsequenceTerms()),
```
@jjgao @inodb this logic is adopted from the annotation pipeline repository, and it only resolves to a single mutation type. I think this is what we want when annotating MAF files. But for hotspot filtering purposes should we consider all consequence terms and return a list of mutation types instead of just one result?

inodb replied:
Yeah after merging #132 we should be able to match on hugo symbol and consequence for any of the transcripts. If one of them matches it's a hotspot

inodb commented on service/src/main/java/org/cbioportal/genome_nexus/service/annotation/CanonicalTranscriptResolver.java:
```diff
@@ -0,0 +1,51 @@
+    {
+        List<TranscriptConsequence> transcripts = new ArrayList<>();
+
+        for (TranscriptConsequence transcript : variantAnnotation.getTranscriptConsequences())
```
I guess this is copied over from the importer but what is canonical differs for mskcc/uniprot/ensembl. Now we are always using canonical ensembl transcript,  if that isn't found the most severe one. Doesn't seem what we want

onursumer replied:
That's right, copied over from importer. With the latest changes to this PR, we are not relying on the canonical transcript anymore when figuring out the hotspots. We may revisit the `CanonicalTranscriptResolver` when we need to use it...

onursumer commented:
Added tests for the hotspot mutation filtering logic. Ideally, we should also test the stuff that we moved over from annotation pipeline. That part is currently not in use for the cancer hotspots endpoint though.

inodb commented on web/src/test/java/org/cbioportal/genome_nexus/web/CancerHotspotsIntegrationTest.java:
```diff
@@ -0,0 +1,191 @@
+            "17,37879794,37879794,G,C", // recurrent
+            "12,25398284,25398285,CC,GA", // recurrent and 3D
+            "12,25380271,25380271,C,T", // 3D
+            "10,89692940,89692940,C,T" // none
```
@onursumer should we handle `chr1` as well?

inodb replied:
things prefixed with `chr*` basically

onursumer replied:
good point, probably we are not handling the `chr*` case right now.

inodb replied:
yeah not a big deal, but would be good to include. Also for indels the format is different for vcf vs maf. So good to have some parameter to indicate which one we are using

inodb commented:
🥞 🌮

inodb commented on web/src/test/java/org/cbioportal/genome_nexus/web/CancerHotspotsIntegrationTest.java:
```diff
@@ -0,0 +1,191 @@
+            "17,37879794,37879794,G,C", // recurrent
+            "12,25398284,25398285,CC,GA", // recurrent and 3D
+            "12,25380271,25380271,C,T", // 3D
+            "10,89692940,89692940,C,T" // none
```
@onursumer what about indels? Splice? Would be good to just indicate those aren't handled yet (if they're not)

onursumer replied:
@inodb indels are handled, see the `testIndelHotspots` method below :) 

Splice is probably not gonna work though

inodb replied:
@onursumer oops missed those

inodb commented on web/src/test/java/org/cbioportal/genome_nexus/web/CancerHotspotsIntegrationTest.java:
```diff
@@ -0,0 +1,191 @@
+        // genomic location strings (needed for GET requests)
+        String[] genomicLocations = {
+            "7,55242466,55242480,GAATTAAGAGAAGCA,-",
+            "7,55249002,55249003,-,CAGCGTGGA"
```
@onursumer this is great! Looks like we are using MAF format here. Maybe good to have some parameter indicating that. 

In the same vain - how are impossible ref/alt/position combos handled?

inodb replied:
ill just filed an issue for my comments so we can merge this.

https://github.com/genome-nexus/genome-nexus/issues/140
